tests/nats/configmap-nats-config.yaml

```
@@ -0,0 +1,24 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nats-config
+data:
+  nats.conf: |
+    pid_file: "/var/run/nats/nats.pid"
+    http: 8222
+
+    cluster {
+      port: 6222
+      routes [
+        nats://nats-0.nats.prod.svc:6222
+        nats://nats-1.nats.prod.svc:6222
+        nats://nats-2.nats.prod.svc:6222
```

As you are deploying nats with replicas set to 1, you'll only get the first one (nats-0).
By the way, the URL embeds the namespace, which is wrong in this context. Let's stick to namespace-local domain names, and use a single server.

```suggestion
        nats://nats-0.nats:6222
```